data/pigui/modules/system-view-ui.lua

```diff
@@ -730,66 +734,78 @@ function Windows.objectInfo.Show()
+			local sma = body.semiMajorAxis
+			local semimajoraxis = nil
+			if sma and sma > 0 then
+				semimajoraxis = ui.Format.Distance(sma)
+			end
+			local rp = body.rotationPeriod * 24 * 60 * 60
+			local op = body.orbitPeriod * 24 * 60 * 60
+			local pop = math.round(body.population * 1e9)
+			local techLevel = starport and SpaceStation.GetTechLevel(body) or nil
+			if techLevel == 11 then
+				techLevel = luc.MILITARY
+			end
+			data = {
+				{ name = lc.MASS, icon = icons.body_radius,
+					value = (not starport) and ui.Format.Mass(body.mass) or nil },
+				{ name = lc.RADIUS, icon = icons.body_radius,
+					value = (not starport) and ui.Format.Distance(body.radius) or nil },
+				{ name = lc.SURFACE_GRAVITY, icon = icons.body_radius,
+					value = (not starport) and ui.Format.Speed(body.gravity, true).." ("..ui.Format.Gravity(body.gravity / 9.8066)..")" or nil },
+				{ name = lc.ORBITAL_PERIOD, icon = icons.body_orbit_period,
+					value = op and op > 0 and ui.Format.Duration(op, 2) or nil },
+				{ name = lc.DAY_LENGTH, icon = icons.body_day_length,
+					value = rp > 0 and ui.Format.Duration(rp, 2) or nil },
+				{ name = luc.ORBIT_APOAPSIS, icon = icons.body_semi_major_axis,
+					value = (parent and not surface) and ui.Format.Distance(body.apoapsis) or nil },
+				{ name = luc.ORBIT_PERIAPSIS, icon = icons.body_semi_major_axis,
+					value = (parent and not surface) and ui.Format.Distance(body.periapsis) or nil },
+				{ name = lc.SEMI_MAJOR_AXIS, icon = icons.body_semi_major_axis,
+					value = semimajoraxis },
+				{ name = lc.ECCENTRICITY, icon = icons.body_semi_major_axis,
+					value = (parent and not surface) and string.format("%0.2f", body.eccentricity) or nil },
+				{ name = lc.AXIAL_TILT, icon = icons.body_semi_major_axis,
+					value = (not starport) and string.format("%0.2f", body.axialTilt) or nil },
+				{ name = lc.POPULATION, icon = icons.personal,
+					value = pop > 0 and ui.Format.NumberAbbv(pop) or nil },
+				{ name = luc.TECH_LEVEL, icon = icons.equipment,
+					value = starport and techLevel or nil }
+			}
+
+		elseif obj.ref:IsShip() then -- physical body
+			-- TODO: the advanced target scanner should add additional data here,
+			-- but we really do not want to hardcode that here. there should be
+			-- some kind of hook that the target scanner can hook into to display
+			-- more info here.
+			-- This is what should be inserted:
+			table.insert(data, { name = luc.SHIP_TYPE, value = body:GetShipType() })
+			if player:GetEquipCountOccupied('target_scanner') > 0 or player:GetEquipCountOccupied('advanced_target_scanner') > 0 then
+				local hd = body:GetEquip("engine", 1)
+				table.insert(data, { name = luc.HYPERDRIVE, value = hd and hd:GetName() or lc.NO_HYPERDRIVE })
+				table.insert(data, { name = luc.MASS, value = Format.MassTonnes(body:GetStats().staticMass) })
+				table.insert(data, { name = luc.CARGO, value = Format.MassTonnes(body:GetStats().usedCargo) })
+			end
+		else
```

Comment:
This section (computing data on a ship target) should not be cached and instead recomputed every update. The values displayed here in the future are expected to change unpredictably (e.g. velocity, signal strength, etc.) and it's easier to pay the small cost of recomputing than to try to invalidate the cached data only when changed.

Reply:
Yes I understand immediate-mode UI is intended for such stuff. I have made the adjustment.